lib/util/cache/package/cacheable.ts

```diff
@@ -0,0 +1,125 @@
+import { DateTime } from 'luxon';
+
+type NotUndefined<T> = T extends undefined ? never : T;
```

Comment:
please add tsdoc annotations to all types and class members.
especially remark which functions are not mutable (return a new object)

Reply:
Actually, I think it's okay modify and return the same object, instead of constructing the new one.

Reply:
yes, but the docs should mention what function is mutating and what is returning a new object